ocaml/xapi/xapi_cluster.ml

```diff
@@ -47,6 +47,50 @@ let create ~__context ~pIF ~cluster_stack ~pool_auto_join ~token_timeout ~token_
       assert_pif_prerequisites (pIF,pifrec);
       let ip = ip_of_pif (pIF,pifrec) in
 
+      let assert_pool_on_same_subnet () =
+        (** we gather ip addresses from all the hosts and also the netmask from the PIF.
+          * if all the ip addresses do not have the same network address, then we raise an
+            error *)
+        let are_on_same_subnet =
```

Comment:
unfortunately i think we need to handle ip4/ip6 separately because the `ipaddr` library doesn't give us an interface which is quite as rich when we have an `Ipaddr.t` (as opposed to an `Ipaddr.V4.t` or `Ipaddr.V6.t`)